### PR TITLE
feat(colors): support new color filtering options

### PR DESCRIPTION
**Pull request by anandaroop**
The type of this PR is: **Feature**

This PR resolves [FX-3719]

### Description

- Adds support for the new colors, while retaining support for old colors, so that this can safely be rolled out during our interim period of indexing both sets of color names
- Once usage of older versions of Eigen drops low enough and we've reindexed all production artworks to new color names only, we can come back and drop support for the `OLD_COLORS` list here

We made some decisions above `foregroundColors` for the new swatches, which determines the color of the check mark in the selected state:

|iOS|Android|
|--|--|
|selected|selected|

Note above that the new colors appear before the old colors, and there is some potential temporary redundancy (blue vs dark blue).

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/main/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [x] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Enable support for filtering new color names - roop

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[FX-3719]: https://artsyproduct.atlassian.net/browse/FX-3719?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 6428d52bdf1518c56f79046fd05ec471af0ea269
</p>

File: src/app/Components/ArtworkFilter/Filters/ColorsOptions.tsx [hunk 1]
```diff
@@ -16,7 +16,8 @@ import React from "react"
 import { ColorsSwatch } from "./ColorsSwatch"
 import { useMultiSelect } from "./useMultiSelect"
 
-export const COLORS = [
+// TODO: delete after all artworks are reindexed to use new colors only
+const OLD_COLORS = [
```
Comment (anandaroop) [inline src/app/Components/ArtworkFilter/Filters/ColorsOptions.tsx:20]: Dropping support for this entirely at this stage (while we are still indexing old color names) resulted in an error, so instead we'll support both versions temporarily, and add this as one of the cleanup tasks in [FX-3777](https://artsyproduct.atlassian.net/browse/FX-3777)

error

**Review by dimatretyak — APPROVED** [review]
LGTM 👍

**Comment by anandaroop** [comment]
> **question**: will `colors` criteria ([link](https://github.com/artsy/eigen/blob/main/data/schema.graphql#L12443)) for saved search alerts be updated too?

There's another related task to update all saved searches to use the new color name as well: [FX-3721](https://artsyproduct.atlassian.net/browse/FX-3721)

Looks like Nikita has already added the code for the migration in https://github.com/artsy/gravity/pull/15058 — but I assume we will only run the migration once we've reindexed all works in production, correct @nickskalkin ?

**Pull request merged**